protocol/amqp/v2/options.go

```diff
@@ -12,42 +12,76 @@ import (
 // Option is the function signature required to be considered an amqp.Option.
 type Option func(*Protocol) error
 
-// WithConnOpt sets a connection option for amqp
-func WithConnOpt(opt amqp.ConnOption) Option {
+type SendOption func(sender *sender)
+type ReceiveOption func(receiver *receiver)
+
+// WithConnOpts sets a connection option for amqp
+func WithConnOpts(opt *amqp.ConnOptions) Option {
```

Comment:
Since you're using pointer semantics in those options, be careful with races - users might not be aware that you're passing the pointer around (instead of copy) and this could lead to surprises for users. I suggest using value semantic to avoid races and these surprises.